src/companion_ad.js

```diff
@@ -1,22 +1,22 @@
-export class CompanionAd {
-  constructor(creativeAttributes = {}) {
-    this.id = creativeAttributes.id || null;
-    this.width = creativeAttributes.width || 0;
-    this.height = creativeAttributes.height || 0;
-    this.assetWidth = creativeAttributes.assetWidth || null;
-    this.assetHeight = creativeAttributes.assetHeight || null;
-    this.expandedWidth = creativeAttributes.expandedWidth || null;
-    this.expandedHeight = creativeAttributes.expandedHeight || null;
-    this.apiFramework = creativeAttributes.apiFramework || null;
-    this.adSlotID = creativeAttributes.adSlotID || null;
-    this.staticResources = [];
-    this.htmlResources = [];
-    this.iframeResources = [];
-    this.adParameters = null;
-    this.xmlEncoded = null;
-    this.altText = null;
-    this.companionClickThroughURLTemplate = null;
-    this.companionClickTrackingURLTemplates = [];
-    this.trackingEvents = {};
-  }
+export function CompanionAd(creativeAttributes = {}) {
```

Comment:
I think we should change the names of all the classes -> functions. They could be something like: `create<Insert name of the class>`, example: `createCompanionAd`

Reply:
Since there are no functions anymore, there are snippets where `instanceof` is used for some checks. Will need to figure out how to handle these. 🤔

Reply:
I would add an `adType` property, to distinguish the variation. But that it is just a suggestion :)